e2e-tests/tests/test_api.py

```diff
@@ -73,32 +76,33 @@ def test_public_api_navigation(self, base_url):
             )
             assert response.status_code == 403
 
-        assert len(signatures) == len(product_versions)
+        assert len(uuid_to_signature) == len(product_versions)
 
         # Check that we can fetch the "signature first date" for each
-        # signature.
+        # signature--some don't have one, so we try to roll with that
         response = requests.get(base_url + '/api/SignatureFirstDate/', {
-            'signatures': signatures.values(),
+            'signatures': uuid_to_signature.values(),
         })
         assert response.status_code == 200
         results = response.json()
-        first_dates = {}
+
+        signature_to_first_date = {}
         for hit in results['hits']:
-            first_dates[hit['signature']] = hit['first_date']
+            signature_to_first_date[hit['signature']] = hit['first_date']
 
-        assert len(first_dates) == len(signatures)
+        assert 0 < len(signature_to_first_date) <= len(uuid_to_signature)
```

Comment:
This is the assertion that was failing and a little bit above, I tweaked the comment about what this section is doing and looking for.

@adngdb Is this in line with what you were thinking?